src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -1,25 +1,69 @@
 package com.es.phoneshop.model.product;
 
-import java.util.List;
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.stream.Collectors;
 
 public class ArrayListProductDao implements ProductDao {
+
+    private static Long counter = 0L;
```

Comment:
Probably remove 'static' from the counter definition. This class is intended to be a singleton, that's why it's not actually needed. This might be the cause of the test failure, as long as for each test the products would have bigger ids (for first test's execution 1-13, then 14-26, etc.)